src/plugins/discover/public/application/components/sidebar/change_indexpattern.tsx

```diff
@@ -104,6 +104,7 @@ export function ChangeIndexPattern({
             onChangeIndexPattern(choice.value);
             setPopoverIsOpen(false);
           }}
+          // @ts-expect-error Property 'value' is missing in type '...'.
```

Comment:
type inferred from `options` is not compatible with `searchProps`

Reply:
Pushed a fix for this to your branch https://github.com/elastic/kibana/pull/83397/commits/e25f7d539f6b45d43e12ed641c6eeed84a895df4